### ISSUE TITLE
Problem with 'One useful value for "PERLDOC_PAGER"' in perldoc.pod

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#11120]
Migrated from [rt.perl.org#83596](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=83596) (status was 'resolved')

Searchable as RT83596$

**Comment by p5pRT** [comment]
### From @scop

#### Created by @scop



perldoc\.pod's ENVIRONMENT section contains&#8203;:

> One useful value for "PERLDOC\_PAGER" is "less \-\+C \-E"\.

However that doesn't work at least with perldoc \-m&#8203;:

$ export PERLDOC\_PAGER="less \-\+C \-E"
$ perldoc \-m Pod&#8203;::Perldoc
\(hit q to exit less\)
Can't exec "less \-\+C \-E"&#8203;: No such file or directory at /usr/share/perl5/Pod/Perldoc\.pm line 1451\.

Not sure if this is a problem with the documentation or the way
perldoc ends up invoking PERLDOC\_PAGER\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=low

This perlbug was built using Perl 5.12.2 in the Fedora build system.
It is being executed now by Perl 5.12.2 - Wed Dec  1 13:04:08 UTC 2010.

Site configuration information for perl 5.12.2:

Configured by Red Hat, Inc. at Wed Dec  1 13:04:08 UTC 2010.

Summary of my perl5 (revision 5 version 12 subversion 2) configuration:
   
  Platform:
    osname=linux, osvers=2.6.32-72.el6.bz634452.x86_64, archname=x86_64-linux-thread-multi
    uname='linux x86-19.phx2.fedoraproject.org 2.6.32-72.el6.bz634452.x86_64 #1 smp fri sep 17 06:52:25 edt 2010 x86_64 x86_64 x86_64 gnulinux '
    config_args='-des -Doptimize=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic -Dccdlflags=-Wl,--enable-new-dtags -DDEBUGGING=-g -Dversion=5.12.2 -Dmyhostname=localhost -Dperladmin=root@localhost -Dcc=gcc -Dcf_by=Red Hat, Inc. -Dprefix=/usr -Dvendorprefix=/usr -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl5 -Dsitearch=/usr/local/lib64/perl5 -Dprivlib=/usr/share/perl5 -Dvendorlib=/usr/share/perl5 -Darchlib=/usr/lib64/perl5 -Dvendorarch=/usr/lib64/perl5 -Darchname=x86_64-linux-thread-multi -Dlibpth=/usr/local/lib64 /lib64 /usr/lib64 -Duseshrplib -Dusethreads -Duseithreads -Dusedtrace=/usr/bin/dtrace -Duselargefiles -Dd_semctl_semun -Di_db -Ui_ndbm -Di_gdbm -Di_shadow -Di_syslog -Dman3ext=3pm -Duseperlio -Dinstallusrbinperl=n -Ubincompat5005 -Uversiononly -Dpager=/usr/bin/less -isr -Dd_gethostent_r_proto -Ud_endhostent_r_proto -Ud_sethostent_r_proto -Ud_endprotoent_r_proto -Ud_setprotoent_r_proto 
 -Ud_endservent_r_proto -Ud_setservent_r_proto -Dscriptdir=/usr/bin -Dotherlibdirs=/usr/local/lib64/perl5/site_perl/5.10.0/x86_64-linux-thread-multi:/usr/local/lib/perl5/site_perl/5.10.0:/usr/lib64/perl5/vendor_perl/5.10.0/x86_64-linux-thread-multi:/usr/lib/perl5/vendor_perl:/usr/lib/perl5/site_perl'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.5.1 20100924 (Red Hat 4.5.1-4)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -fstack-protector'
    libpth=/usr/local/lib64 /lib64 /usr/lib64
    libs=-lresolv -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lpthread -lc -lgdbm_compat
    perllibs=-lresolv -lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.12.90'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,--enable-new-dtags -Wl,-rpath,/usr/lib64/perl5/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic'

Locally applied patches:
    


@INC for perl 5.12.2:
    /usr/local/lib64/perl5
    /usr/local/share/perl5
    /usr/lib64/perl5
    /usr/share/perl5
    /usr/lib64/perl5
    /usr/share/perl5
    /usr/local/lib64/perl5/site_perl/5.10.0/x86_64-linux-thread-multi
    /usr/local/lib/perl5/site_perl/5.10.0
    /usr/lib64/perl5/vendor_perl/5.10.0/x86_64-linux-thread-multi
    /usr/lib/perl5/vendor_perl
    /usr/lib/perl5/site_perl
    .


Environment for perl 5.12.2:
    HOME=/home/scop
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/lib64/ccache:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/NX/bin:/home/scop/bin:/usr/NX/bin
    PERLDOC_PAGER=less -+C -E
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Mon Feb 07 12&#8203;:10&#8203;:38 2011\, scop wrote&#8203;:

> 
> perldoc\.pod's ENVIRONMENT section contains&#8203;:
> 
> > One useful value for "PERLDOC\_PAGER" is "less \-\+C \-E"\.
> 
> However that doesn't work at least with perldoc \-m&#8203;:
> 
> $ export PERLDOC\_PAGER="less \-\+C \-E"
> $ perldoc \-m Pod&#8203;::Perldoc
> \(hit q to exit less\)
> Can't exec "less \-\+C \-E"&#8203;: No such file or directory at
>    /usr/share/perl5/Pod/Perldoc\.pm line 1451\.
> 
> Not sure if this is a problem with the documentation or the way
> perldoc ends up invoking PERLDOC\_PAGER\.
> 

I can confirm that I get this interaction when using 'perldoc \-m'\.

$ echo $PERLDOC\_PAGER
less \-\+C \-E

$ perldoc \-m Pod&#8203;::Perldoc
Can't exec "less  \-\+C \-E"&#8203;: No such file or directory at
/usr/local/lib/perl5/5\.14\.2/Pod/Perldoc\.pm line 1451\.
Can't exec "/usr/bin/less \-R"&#8203;: No such file or directory at
/usr/local/lib/perl5/5\.14\.2/Pod/Perldoc\.pm line 1451\.
\[\.\.\. and then the code displays\]

\.\.\. but I should note that I don't get it with simple 'perldoc' or with
'perldoc \-l'\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From zengargoyle@gmail.com

On Fri\, Dec 9\, 2011 at 6&#8203;:24 PM\, James E Keenan via RT
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> On Mon Feb 07 12&#8203;:10&#8203;:38 2011\, scop wrote&#8203;:
>>
>> perldoc\.pod's ENVIRONMENT section contains&#8203;:
>>
>> > One useful value for "PERLDOC\_PAGER" is "less \-\+C \-E"\.
>>
>> However that doesn't work at least with perldoc \-m&#8203;:
>>
>> $ export PERLDOC\_PAGER="less \-\+C \-E"
>> $ perldoc \-m Pod&#8203;::Perldoc
>> \(hit q to exit less\)
>> Can't exec "less \-\+C \-E"&#8203;: No such file or directory at
>>    /usr/share/perl5/Pod/Perldoc\.pm line 1451\.
>>
>> Not sure if this is a problem with the documentation or the way
>> perldoc ends up invoking PERLDOC\_PAGER\.
>>
>
> I can confirm that I get this interaction when using 'perldoc \-m'\.
>
> $ echo $PERLDOC\_PAGER
> less \-\+C \-E
>
> $ perldoc \-m Pod&#8203;::Perldoc
> Can't exec "less  \-\+C \-E"&#8203;: No such file or directory at
> /usr/local/lib/perl5/5\.14\.2/Pod/Perldoc\.pm line 1451\.
> Can't exec "/usr/bin/less \-R"&#8203;: No such file or directory at
> /usr/local/lib/perl5/5\.14\.2/Pod/Perldoc\.pm line 1451\.
> \[\.\.\. and then the code displays\]
>
> \.\.\. but I should note that I don't get it with simple 'perldoc' or with
> 'perldoc \-l'\.

page\_module\_file\(\) uses system\($pager\, @&#8203;found\)
page\(\) uses system\("$pager $output"\)

I'm guessing it would need either shell quoting @&#8203;found so that they can be
safely uses in a one argument system call\, or checking and splitting $pager
into the executable and arguments for the multi argument system call\.



---
Carl

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri Dec 09 19&#8203;:52&#8203;:28 2011\, zengargoyle@&#8203;gmail\.com wrote&#8203;:

More data points\. Running with debugging\, my output \(STDERR?\) started
with this&#8203;:

\#\#\#
$ export PERLDOC\_PAGER="less \-\+C \-E"
$ perldoc \-D \-m lib/Pod/Perldoc\.pm
Ending switch processing\.  Args are \[lib/Pod/Perldoc\.pm\] with 0 errors\.
/usr/local/bin/perldoc => Pod&#8203;::Perldoc v3\.15\_04

Formatter class Pod&#8203;::Perldoc&#8203;::ToMan successfully loaded\!
Will format with the class Pod&#8203;::Perldoc&#8203;::ToMan
Searching for lib/Pod/Perldoc\.pm
Found as lib/Pod/Perldoc\.pm
About to try calling less \-\+C \-E lib/Pod/Perldoc\.pm
Can't exec "less \-\+C \-E"&#8203;: No such file or directory at
/usr/local/lib/perl5/5\.14\.2/Pod/Perldoc\.pm line 1451\.
That didn't work\.
About to try calling /usr/bin/less \-R lib/Pod/Perldoc\.pm
Can't exec "/usr/bin/less \-R"&#8203;: No such file or directory at
/usr/local/lib/perl5/5\.14\.2/Pod/Perldoc\.pm line 1451\.
That didn't work\.
About to try calling more lib/Pod/Perldoc\.pm
\#\#\#

\['more' worked; program exited with&#8203;:
\#\#\#
Yay\, it worked\.
\#\#\#

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri Dec 09 19&#8203;:52&#8203;:28 2011\, zengargoyle@&#8203;gmail\.com wrote&#8203;:
> 
> page\_module\_file\(\) uses system\($pager\, @&#8203;found\)
> page\(\) uses system\("$pager $output"\)
> 
> I'm guessing it would need either shell quoting @&#8203;found so that they can be
> safely uses in a one argument system call\, or checking and splitting
$pager
> into the executable and arguments for the multi argument system call\.
> 

This appears to be very similar to the problem described in
https://rt-archive.perl.org/perl5/Ticket/Display.html?id=22329.&nbsp;  Should this
ticket be merged into that older one?

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
### From @rjbs

This appears to be fixed by the latest Pod&#8203;::Perldoc\, which is in blead\.

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

This appears to be fixed by the latest Pod&#8203;::Perldoc\, which is in blead\.

**Comment by p5pRT** [comment]
@rjbs - Status changed from 'open' to 'resolved'
